app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/welcome/WelcomeOnBoardScreen.kt

```diff
@@ -13,79 +14,125 @@ import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.semantics.semantics
 import androidx.compose.ui.semantics.testTag
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.arygm.quickfix.R
+import com.arygm.quickfix.model.offline.small.PreferencesViewModel
 import com.arygm.quickfix.ressources.C
 import com.arygm.quickfix.ui.elements.QuickFixButton
 import com.arygm.quickfix.ui.navigation.NavigationActions
 import com.arygm.quickfix.ui.theme.poppinsTypography
 import com.arygm.quickfix.ui.userModeUI.navigation.UserScreen
 
 @Composable
-fun WelcomeOnBoardScreen(navigationActions: NavigationActions) {
+fun WelcomeOnBoardScreen(
+    navigationActions: NavigationActions,
+    preferencesViewModel: PreferencesViewModel,
+    testingFlag: Boolean = false
+) {
   BoxWithConstraints {
     val widthRatio = maxWidth / 411
     val heightRatio = maxHeight / 860
-    Column(
-        modifier = Modifier.fillMaxSize().padding(16.dp),
-        horizontalAlignment = Alignment.CenterHorizontally,
-    ) {
-      Spacer(modifier = Modifier.height(100.dp * heightRatio.value))
-      Text(
-          "Welcome on board !!",
-          style =
-              poppinsTypography.headlineLarge.copy(
-                  color = MaterialTheme.colorScheme.onBackground, fontSize = 26.sp))
-      Spacer(modifier = Modifier.weight(0.2f))
-      Row(modifier = Modifier.weight(0.9f)) {
-        Image(
-            painter =
-                painterResource(
-                    id =
-                        R.drawable
-                            .onboarding_worker_1), // Replace 'my_image' with your PNG file name
-            contentDescription = "Description of the image", // Accessibility description
-            contentScale = ContentScale.Crop,
-            modifier =
-                Modifier.fillMaxSize().semantics {
-                  testTag = C.Tag.welcomeOnBoardScreenImage
-                } // Optional: Set how the image should scale
-            )
+    val isWorker by preferencesViewModel.isWorkerFlow.collectAsState(initial = false)
+
+    // If the user chooses "Switch Worker Mode," we set this to true and show a loader until
+    // isWorker == true
+    var isWaitingForWorkerMode by remember { mutableStateOf(false) }
+
+    // When isWorker changes, if we're waiting for worker mode and it's now true, navigate
+    LaunchedEffect(isWorker) {
+      if (isWaitingForWorkerMode && isWorker) {
+        navigationActions.navigateTo(UserScreen.PROFILE)
```

Comment:
It's really nice to have `isWaitingForWorkerMode` variable to display a loader when the user switches to worker mode, but don't you think you should set it to false after successfully navigating in worker mode?